semgrep/semgrep/target_manager_extensions.py

```diff
@@ -62,30 +73,64 @@
 GENERIC_EXTENSIONS = [FileExtension("")]
 
 _LANGS_TO_EXTS: Dict[Language, List[FileExtension]] = {
-    Language.PYTHON: PYTHON_EXTENSIONS,
-    Language.PYTHON2: PYTHON_EXTENSIONS,
-    Language.PYTHON3: PYTHON_EXTENSIONS,
-    Language.JAVASCRIPT: JAVASCRIPT_EXTENSIONS,
-    Language.TYPESCRIPT: TYPESCRIPT_EXTENSIONS,
-    Language.JAVA: JAVA_EXTENSIONS,
     Language.C: C_EXTENSIONS,
+    Language.CSHARP: CSHARP_EXTENSIONS,
+    Language.GENERIC: GENERIC_EXTENSIONS,
     Language.GO: GO_EXTENSIONS,
-    Language.ML: ML_EXTENSIONS,
-    Language.RUBY: RUBY_EXTENSIONS,
-    Language.PHP: PHP_EXTENSIONS,
     Language.HACK: HACK_EXTENSIONS,
+    Language.HCL: HCL_EXTENSIONS,
+    Language.HTML: HTML_EXTENSIONS,
+    Language.JAVA: JAVA_EXTENSIONS,
+    Language.JAVASCRIPT: JAVASCRIPT_EXTENSIONS,
     Language.JSON: JSON_EXTENSIONS,
-    Language.LUA: LUA_EXTENSIONS,
-    Language.CSHARP: CSHARP_EXTENSIONS,
-    Language.RUST: RUST_EXTENSIONS,
     Language.KOTLIN: KOTLIN_EXTENSIONS,
-    Language.YAML: YAML_EXTENSIONS,
+    Language.LUA: LUA_EXTENSIONS,
+    Language.ML: ML_EXTENSIONS,
+    Language.PHP: PHP_EXTENSIONS,
+    Language.PYTHON2: PYTHON_EXTENSIONS,
+    Language.PYTHON3: PYTHON_EXTENSIONS,
+    Language.PYTHON: PYTHON_EXTENSIONS,
     Language.REGEX: GENERIC_EXTENSIONS,
-    Language.GENERIC: GENERIC_EXTENSIONS,
+    Language.RUBY: RUBY_EXTENSIONS,
+    Language.RUST: RUST_EXTENSIONS,
     Language.SCALA: SCALA_EXTENSIONS,
+    Language.TYPESCRIPT: TYPESCRIPT_EXTENSIONS,
     Language.VUE: VUE_EXTENSIONS,
-    Language.HTML: HTML_EXTENSIONS,
-    Language.HCL: HCL_EXTENSIONS,
+    Language.YAML: YAML_EXTENSIONS,
+}
+
+#
+# Define for which languages we allow executable, extensionless files to
+# be passed to semgrep-core. Semgrep-core will inspect the beginning of
+# the file to determine if it's the requested language.
+#
+# Language.GENERIC is special. Semgrep-core will accept all the files if
+# the language is 'generic'.
+#
+_LANG_IS_SCRIPT: Dict[Language, bool] = {
+    Language.C: False,
+    Language.CSHARP: False,
+    Language.GENERIC: False,  # important/special
+    Language.GO: False,
+    Language.HACK: True,
+    Language.HTML: False,
+    Language.JAVA: False,
+    Language.JAVASCRIPT: True,
+    Language.JSON: False,
+    Language.KOTLIN: False,
+    Language.LUA: True,
+    Language.ML: True,
+    Language.PHP: True,
+    Language.PYTHON2: True,
+    Language.PYTHON3: True,
+    Language.PYTHON: True,
+    Language.REGEX: False,  # special
+    Language.RUBY: True,
+    Language.RUST: True,
+    Language.SCALA: True,
+    Language.TYPESCRIPT: True,
+    Language.VUE: False,
+    Language.YAML: False,
 }
```

Comment:
This is for `lang_is_script`, which is unused for now. I did that earlier, but didn't finish because it's too much for one PR.